sdk/storage/storage-blob/src/policies/SharedKeyCredentialPolicy.ts

```diff
@@ -76,6 +76,13 @@ export class SharedKeyCredentialPolicy extends CredentialPolicy {
       `SharedKey ${this.factory.accountName}:${signature}`
     );
 
+    // Workaround for https://github.com/axios/axios/issues/2107
+    // We should always keep the 'content-length' header once the issue is solved
+    // For a better explanation about this workaround, look here: https://github.com/Azure/azure-sdk-for-js/pull/3273
+    if (typeof request.body !== "function" && !(request.body && request.onUploadProgress)) {
```

Comment:
Any specific reason to check onUploadProgress and "function" type? Why not just remove line #50 ~ #52?

Reply:
A detailed explanation can be found here: #3273